crates/test-programs/wasi-tests/src/lib.rs

```diff
@@ -23,3 +23,28 @@ pub fn open_scratch_directory(path: &str) -> Result<wasi_unstable::Fd, String> {
         Ok(dir_fd)
     }
 }
+
+pub fn open_scratch_directory_new(path: &str) -> Result<wasi::Fd, String> {
```

Comment:
I fear this might lead to confusion in the future when adding new tests, where one has to consciously remember which version of `open_scratch_directory` to choose. Regardless though I reckon this is the best we can do at least for now. Perhaps, for the future, we could consider feature-gating and some clever use of `cfg` attributes?

Reply:
Yeah this isn't great unfortunately, I wrote up a bit more about it [here](https://github.com/bytecodealliance/wasmtime/issues/632#issuecomment-562282359) about why it's necessary.

I'd hope though that after this PR the transition for the remaining tests would happen in pretty short order which would allow us to delete the previous version. I can add comments though if you'd like about how it's purely a transitionary thing.